amdctl.c

```diff
@@ -89,6 +89,16 @@ static uint64_t buffer;
 static int PSTATES = 8, DIDS = 5, cpuFamily = 0, cpuModel = -1, cores = 0,
 		pvi = 0, debug = 0, quiet = 0, testMode = 0, core = -1, pstate = -1;
 
+// AMD14H (Bobcat) related constants and static vars
+#define REG_CLOCK_POWER_CONTROL	      0xd4
+#define COFVID_MIN_VID_BITS           "48:42"
+#define COFVID_MAX_VID_BITS           "41:35"
+static int COFVID_MAX_VID	      =	1;
+static int COFVID_MIN_VID	      = 128;
+static int mainPllCof                 = -1;
+static char *ADDR_CLOCK_POWER_CONTROL =	"18.3";
+static char *MAIN_PLL_OP_FREQ_ID_BITS = "5:0";
```

Comment:
Could this and ADDR_CLOCK_POWER_CONTROL be constants? They don't seem to change.

Reply:
I think ADDR_CLOCK_POWER_CONTROL and MAIN_PLL_OP_FREQ_ID_BITS can be constants, while the rest is set in a block around line 488:
```
mainPllCof = 100 * (getDec(MAIN_PLL_OP_FREQ_ID_BITS) + 16);
core = 0;
getReg(COFVID_STATUS);
COFVID_MIN_VID = getDec(COFVID_MIN_VID_BITS);
COFVID_MAX_VID = getDec(COFVID_MAX_VID_BITS);
```

I think during development I had them as constants, but then ran into compiler warning and errors... might be worth a try.